include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -929,6 +1112,8 @@ namespace VectorTools
                 const hp::QCollection<dim-1>               &q_boundary,
                 const bool                                  project_to_boundary_first)
   {
+    Assert((dynamic_cast<const parallel::distributed::Triangulation<dim,spacedim>* > (&(dof.get_triangulation()))==0),
```

Comment:
Yes, but I guess it's a good idea to fail with something more descriptive than `Can't ask for DoF indices on artificial cells.` if you accidentally use `project` instead of `project_distributed`.